api/src/pcapi/routes/auth/discord.py

```diff
@@ -1,14 +1,100 @@
+import datetime
+from urllib import parse as urlparse
+from urllib.parse import urlencode
+
+from flask import redirect
 from flask import render_template
 from flask import request
+from werkzeug.wrappers.response import Response
 
+from pcapi import settings
+from pcapi.app import csrf
+from pcapi.core import token as token_utils
+from pcapi.core.users import exceptions as users_exceptions
+from pcapi.core.users import models as user_models
+from pcapi.core.users import repository as users_repo
+from pcapi.models import db
 from pcapi.routes.auth.forms.forms import SigninForm
 
 from . import blueprint
 
 
+def process_redirect_url(redirect_url: str, authorization_code: str) -> str:
+    # Prepare the redirect URL
+    url_parts = list(urlparse.urlparse(redirect_url))
+    queries = dict(urlparse.parse_qsl(url_parts[4]))
+    queries.update({"authorization_code": authorization_code})
```

Comment:
[style] Plus court : 
```suggestion
    queries["authorization_code"] = authorization_code
```

(Au passage, je pense qu'on pourrait nommer la variable `query`, comme dans _query string_. Mais je chipote. :) )